staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_json.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/json"
+)
+
+// JSON represents any valid JSON value.
+// These types are supported: bool, int64, float64, string, []interface{}, map[string]interface{} and nil.
+//
+// +protobuf=true
+// +protobuf.options.marshal=false
+// +protobuf.as=ProtoJSON
+// +protobuf.options.(gogoproto.goproto_stringer)=false
+// +k8s:conversion-gen=false
+type JSON struct {
+	Object interface{} `json:"-" protobuf:"-"`
+}
```

Comment:
do we have any sense how disruptive this will be to consumers of these Go types (e.g. https://grep.app/search?q=.JSON%7BRaw&case=true&filter[lang][0]=Go)